src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -359,75 +441,129 @@ async def _handle_unowned_imports(
         raise UnownedDependencyError(msg)
 
 
-@rule(desc="Inferring Python dependencies by analyzing source")
-async def infer_python_dependencies_via_source(
-    request: InferPythonImportDependencies,
+@rule
+async def _exec_parse_deps(
```

Comment:
If this is purely an internal API called to produce `ResolvedParsedPythonDependencies`, then ditto potentially `@_rule_helper`?